forwardproxy.go

```diff
@@ -109,12 +109,16 @@ func dualStream(w1 io.Writer, r1 io.Reader, w2 io.Writer, r2 io.Reader) error {
 
 	go stream(w1, r1)
 	go stream(w2, r2)
-	err1 := <-errChan
-	err2 := <-errChan
-	if err1 != nil {
-		return err1
+
+	firstHangerErr := <-errChan
```

Comment:
What's a hanger?

Reply:
There are 2 connections, that spawn 2 goroutines
1) relay connection1 -> connection2
2) relay connection2 -> connection1
whoever hungs up first, gets their error written into firstHangerErr. That part hasn't really changed, I just renamed the variable. But what I also realized is that there is not point in returning other side's error. Pretty simple, but I can add a comment or something.
What happens next is that we either wait for the other side to close and write error into `errChan`, or we timeout, leave the function, and defers close everything unclosed.